src/engine/evaluate.ts

```diff
@@ -7,18 +7,33 @@ import {
   AudienceDefinitionFilter,
 } from '../../types';
 
-export const evaluateCondition = (
-  condition: EngineCondition<AudienceDefinitionFilter>,
-  pageViews: PageView[]
-): boolean => {
-  const { filter, rules } = condition;
-
-  const filteredPageViews = filter.queries.flatMap((query) =>
-    pageViews.filter((pageView) => {
+/* Checks each pageView once for any matching queries
+ * and returns the filtered array containing the matched
+ * pageViews
+ */
+const filterPageViews = (
+  filter: EngineCondition<AudienceDefinitionFilter>['filter'],
+  pageViews: Readonly<PageView[]>
+): PageView[] => {
+  return pageViews.filter((pageView) =>
+    filter.queries.some((query) => {
       const pageFeatures = pageView.features[query.queryProperty];
       return queryMatches(query, pageFeatures);
     })
   );
+};
```

Comment:
Here, also, we have an opportunity to implement the internal `AND` logic, swapping `some` for `every` according to the value of `filter.any`.

Reply:
Cool!

lets add that as a note?